internal/provider/resource_api_key.go

```diff
@@ -0,0 +1,141 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func resourceAPIKey() *schema.Resource {
+	return &schema.Resource{
+		Description:   "A key to access the Neon API.",
+		SchemaVersion: 7,
+		CreateContext: resourceAPIKeyCreateRetry,
+		ReadContext:   resourceAPIKeyReadRetry,
+		DeleteContext: resourceAPIKeyDeleteRetry,
+		Schema: map[string]*schema.Schema{
+			"key_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the API Key.",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The API key ID.",
+			},
+			"key": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+				Description: "The generated 64-bit token required to access the Neon API.",
+			},
+		},
+	}
+}
+
+type apiKey struct {
+	id      string
+	key     string
+	keyName string
+}
```

Comment:
**(question)** Is this type required? Why not use the SDK type directly? Is there a particular reason to maintain this other than ability to swap Go SDK with other adapter to the Neon API?

Reply:
As far as I can tell, there is no type that would match something like this in the SDK (unlike say Role or Endpoint), the closest is [ApiKeyCreateResponse](https://pkg.go.dev/github.com/kislerdm/neon-sdk-go#ApiKeyCreateResponse), would you suggest to use that instead? Happy to do so, it just felt like in comparison with other resources there were no equivalent in the SDK 🙂

Reply:
@mickael-carl it's ok, let's keep a dedicate type.

Also, I figured that I missed a few points 🤦🏻 - see more comments. Sorry for sloppiness 😄